metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1610,6 +1640,31 @@ class MetalsLanguageServer(
     )
   }
 
+  private def onWorksheetChangedUnbatched(
```

Comment:
Can we move this inside the WorksheetProvider ?

Reply:
I simplified this implementation a bit. For consistency with `onBuildChanged` I'd like to keep it in this file for now